marconi-core/src/Marconi/Core/Coordinator.hs

```diff
@@ -141,17 +142,16 @@ processQueue f initialState q cBox =
 step
   :: ( Ord (Point input)
      , IsIndex m input indexer
-     , MonadError IndexerError m
      )
   => indexer input
   -> ProcessedInput (Point input) input
-  -> m (indexer input)
+  -> ExceptT IndexerError m (indexer input)
```

Comment:
I left it like this because this function doesn't demand `IO`, and I figured we should be using `ExceptT`/`Either` in non-`IO` functions (which ties into my idea about thinning out the `IO` layer)